File: typescript/src/messages/verdict_response.ts
import { JsonProperty, JsonObject } from "typescript-json-serializer";
import { Kind, Message } from "./message";
import { Verdict } from "../Verdict";

@JsonObject()
export class VerdictResponse extends Message {
  public constructor(
    @JsonProperty() public sha256: string,
    @JsonProperty() public guid: string,
    @JsonProperty() public verdict: Verdict,
    @JsonProperty() public upload_token: string | undefined,
    @JsonProperty() public url: string | undefined
  ) {
    super(Kind.VerdictResponse);
  }
}

File: go/vaas/pkg/messages/verdict_response.go
package messages

import (
	"net/url"
)

type VerdictResponse struct {
	Kind        string  `json:"kind"`
	Sha256      string  `json:"sha256"`
	Guid        string  `json:"guid"`
	Verdict     Verdict `json:"verdict"`
	Url         url.URL `json:"url"`
	UploadToken string  `json:"upload_token"`
}

func (response VerdictResponse) IsValid() bool {
	return response.Sha256 != "" && response.Verdict != ""
}
